### ISSUE TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'call')

### ISSUE THREAD

**Issue by wasif9** [highcharts/highcharts#18941]
I am trying to save and load annotations to and from the local storage. I have a few custom annotations that I want to save and load. Its storing and loading correctly as suggested by the browser console. the formatting is correct and all the data needed to redraw the annotation is being stored and loaded as well. however It doesnt get drawn on the chart and throws  the following error:
Uncaught TypeError: Cannot read properties of undefined (reading 'call')
    at h.redraw (annotations-advanced.src.js:1842:31)
    at annotations-advanced.src.js:4143:31
    at Array.forEach (<anonymous>)
    at e.redraw (annotations-advanced.src.js:4143:31)
    at e.redraw (annotations-advanced.src.js:5969:17)
    at g.redrawItem (annotations-advanced.src.js:10:2)
    at g.redrawItems (annotations-advanced.src.js:10:2)
    at g.redraw (annotations-advanced.src.js:10:2)
    at annotations-advanced.src.js:10:2
    at Array.forEach (<anonymous>)
The stock-tools by highcharts is being loaded and stored correctly but I cant figure out why custom annotations arent being stored and loaded correctly.

here is one of the custom annotations
const supportLineStart = (e, userOptions) => {
    var coords = this.chart.pointer.getCoordinates(e),
        navigation = this.chart.options.navigation,
        LineMode = 'Support',
        labelPosition;
    controlPoints = [{
        style: { // control point styling 
            fill: 'rgba(20, 200, 20, 1)',
            stroke: 'rgba(20, 200, 20, 1)'
        },
        positioner: function (target) { // is called with any movement 
            var xy = Highcharts.Annotation.MockPoint.pointToPixels(
                target.points[1]
            ); //creates mock point at point 1
            startCoords = target.chart.yAxis[0].toPixels(coords.yAxis[0].value);
            labelPosition = target.chart.yAxis[0].toValue(xy.y);
            return {
                x: xy.x - this.graphic.width / 2,
                y: xy.y - this.graphic.height / 2
            };

        },
        events: {
            // will change support line type when second point is clicked
            click: function (e, target) {
                if (LineMode === 'Support') {
                    LineMode = 'Minor Support';
                    target.annotation.shapes[0].update({
                        stroke: 'rgba(20, 200, 20, 0.75)',
                        dashStyle: 'ShortDash'
                    });
                    target.annotation.labels[0].update({
                        style: {
                            color: 'rgb(20, 200, 20)',
                            fontWeight: 'normal',
                        },
                        text: LineMode,
/*                        point: function (annotation) {
                            return {
                                xAxis: annotation.chart.xAxis[0].max,
                                yAxis: 0,
                                //convert 
                                x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 16,
                                y: labelPosition
                            };
                        },*/

                        point: {
                            xAxis: annotation.chart.xAxis[0].max,
                            yAxis: 0,
                            x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 16,
                            y: labelPosition,
                            labelPosition: labelPosition
                        },
                    });
                } else if (LineMode === 'Prior Support') {
                    LineMode = 'Support';
                    //Will make changes to current  
                    target.annotation.shapes[0].update({
                        stroke: 'rgba(20, 200, 20, 0.75)',
                        dashStyle: 'Solid'
                    });
                    target.annotation.labels[0].update({
                        style: {
                            color: 'rgb(20, 200, 20)',
                            fontWeight: 'Bold',
                        },
                        text: LineMode,
                        // points have to be changed to position label correctly
/*                        point: function (annotation) {
                            return {
                                xAxis: annotation.chart.xAxis[0].max,
                                yAxis: 0,
                                x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 30,
                                y: labelPosition
                            };
                        },*/
                        point: {
                            xAxis: annotation.chart.xAxis[0].max,
                            yAxis: 0,
                            x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 30,
                            y: labelPosition,
                            labelPosition: labelPosition

                        },
                    });
                } else if (LineMode === 'Minor Support') {
                    LineMode = 'Prior Support';
                    target.annotation.shapes[0].update({
                        stroke: 'rgba(0, 0, 0, 0.75)',
                        dashStyle: 'ShortDot'
                    });
                    target.annotation.labels[0].update({
                        style: {
                            color: 'rgb(0, 0, 0)',
                            fontWeight: 'normal',
                        },
                        text: LineMode,
/*                        point: function (annotation) {
                            return {
                                xAxis: annotation.chart.xAxis[0].max,
                                yAxis: 0,
                                x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 18,
                                y: labelPosition
                            };
                        },*/

                        point: {
                            xAxis: annotation.chart.xAxis[0].max,
                            yAxis: 0,
                            x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 18,
                            y: labelPosition,
                            labelPosition: labelPosition

                        },
                    });
                }
            }
        }
    }, {
        positioner: function (target) { // is called with any movement 
            var xy = Highcharts.Annotation.MockPoint.pointToPixels(
                target.points[0]
            ); //creates mock point at point 0
            return {
                x: xy.x - this.graphic.width / 2,
                y: xy.y - this.graphic.height / 2
            };

        },
        events: { // allows the 
            drag: function (e, target) {
                target.translatePoint(
                    e.chartX - (
                        target.points[0].plotX + target.chart.plotLeft
                    ),
                    0,
                    0
                );
                target.redraw(false);
            },
        }
    }];
    this.chart.addAnnotation(Highcharts.merge({
        langKey: 'Support Line',
        draggable: 'y',
        labelOptions: {
            backgroundColor: 'rgba(255, 255, 255, 0)',
            borderColor: 'rgba(255, 255, 255, 0)',
        },

        labels: [{
            allowOverlap: false,
            verticalAlign: 'middle',
            distance: -11,

/*            point: function (annotation) {
                return {
                    xAxis: annotation.chart.xAxis[0].max,
                    yAxis: 0,
                    x: annotation.chart.xAxis[0].toPixels(annotation.chart.xAxis[0].max) - 30,
                    y: labelPosition //makes it so the label moves with the line 
                };
            },*/

            point: {
                xAxis: this.chart.xAxis[0].max,
                yAxis: 0,
                x: this.chart.xAxis[0].toPixels(this.chart.xAxis[0].max) - 30,
                y: labelPosition,
                labelPosition: labelPosition
            },

            overflow: 'allow',
            style: {
                color: 'rgb(20, 200, 20)',
                fontWeight: 'bold',
            },
            text: LineMode,
        }],
        shapes: [{
            dashStyle: 'Solid',
            type: 'path',
            stroke: 'rgba(20, 200, 20, 0.75)',
            strokeWidth: 2,
            points: [{
                xAxis: 0,
                yAxis: 0,
                x: coords.xAxis[0].value,
                y: coords.yAxis[0].value
            }, {
                xAxis: 0,
                yAxis: 0,
                x: this.chart.xAxis[0].getExtremes().max,
                y: coords.yAxis[0].value
            }],
            controlPoints: controlPoints
        }],
    }, navigation
        .annotationsOptions, navigation.bindings.rectangleAnnotation.annotationsOptions));

    return true;
}

**Comment by pawellysy** [comment]
Hi @wasif9, thanks for reaching out to us!
Could you reproduce the issue in the online editor?
Since you are using react, you can use this demo as a starting point: https://codesandbox.io/s/highcharts-react-demo-e1ue3

Thanks!

**Comment by wasif9** [comment]
Hi, I am unable to reproduce the issue however I can walk you through what I did, what the issue is and at what point the code is breaking. Let me know and I will provide the relevant snippets and the error

**Comment by wasif9** [comment]
I have the custom annotations defined  saving them and loading them works correctly  as the formats of the objects match. The chart.addAnnotation and possibly redraw functions break when I try to add it to the chart. I have tried almost everything. Removing the annotations before adding, using different functions for the custom annotations to save and load but all of them yields the same issue i mentioned above. I have tried cloning a highcharts stock-tool annotation which seems to work and get redrawn perfectly and i made incremental changes to see where it breaks. Labels work fine and ive tried customizing them. When i try to draw a custom path annotation that is supposed to work like a support line, the annotation gets drawn at first. Then i save it and load it back. The objects saved and retrieved are same and have the same format but the load function also has  chart = Highcharts.stockChart('container', JSON.parse(localStorage.customStockToolsChart)); and this is where i get the error above. every effort in trying to debug it pointing towards the highcharts library js files. Would highly appreciate the assistance as its getting pretty annoying.
